aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -643,22 +647,42 @@ Tensor _mkldnn_convolution_transpose(
     TORCH_CHECK(mkldnn_bf16_device_check(),
         "mkldnn_convolution_transpose: bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
   }
-  bool is_channels_last = input_t.suggest_memory_format() == at::MemoryFormat::ChannelsLast;
 
-  auto output_sizes = conv_input_size(input_t.sizes(), weight_t.sizes(), padding, output_padding, stride, dilation, groups);
-  auto output = at::empty({0}, input_t.options());
+  std::vector<int64_t> weight_IOHW_sizes = _original_deconv_weight_size(weight_t, groups);
 
-  const ideep::tensor x = itensor_from_tensor(input_t);
-  ideep::tensor w = itensor_from_tensor(weight_t);
-  // mkldnn transposed convolution has weight in logical order of OIHW or OIDHW,
-  // while PyTorch has IOHW or IODHW, `._tranpose()` switches strides (no memory copy).
-  w.transpose_(0, 1);
+  auto memory_format =
+      mkldnn_convolution_memory_format(input_t.ndimension(), use_channels_last);
+
+  auto input = input_t.is_mkldnn() ? input_t : input_t.contiguous(memory_format);
+  auto weight = weight_t.is_mkldnn() ? weight_t : weight_t.contiguous(memory_format);
+
+  auto output_sizes = conv_input_size(input.sizes(), weight_IOHW_sizes, padding, output_padding, stride, dilation, groups);
+  auto output = at::empty({0}, input.options());
+
+  const ideep::tensor x = itensor_from_tensor(input);
+
+  ideep::tensor w;
+  if (weight.is_mkldnn()) {
+    // weight is the prepacked mkldnn tensor
+    w = itensor_from_tensor(weight);
+  } else {
+    // aten tensor with the shape of the prepacked tensor
+    at::Tensor origin_weight_t;
+    if (groups > 1) {
+      origin_weight_t = weight.transpose(0, 1).reshape(weight_IOHW_sizes);
+    } else {
+      origin_weight_t = weight.transpose(0, 1);
+    }
+    w = itensor_from_tensor(origin_weight_t);
+    w.transpose_(0, 1);
```

Comment:
Do we have a case that we want to handle this "aten tensor with the shape of the prepacked tensor", or we always model prepacked weight with "MKLDNNTensor"?

Reply:
When compiling the graph, each node is run with FakeTensor inputs and in this case, the weight tensor is an aten tensor with the shape of the prepacked tensor.

Reply:
For Meta tensor: added an implementation `mkldnn_convolution_transpose_pointwise_meta`.

For MKLDNN and CPU tensor: supported the computation for both of them in the original implementation: `mkldnn_convolution_transpose_pointwise`.